mis/pipeline/preprocessor.py

```diff
@@ -0,0 +1,46 @@
+import abc
+
+import networkx as nx
+
+
+class BasePreprocessor(abc.ABC):
+    """
+    Base class for preprocessors.
+    """
+
+    @abc.abstractmethod
+    def preprocess(self, graph: nx.Graph) -> nx.Graph: ...
+
+    """
+    Preprocess a graph.
+
+    Typically, this will do two things:
+
+    1. apply transformations to the graph to reduce its size
```

Comment:
Would it make sense to add a `reduce` method for this?

Reply:
`Fixture` is in charge of calling `preprocess` and `rebuild`.